app/controllers/course_summaries_controller.rb

```diff
@@ -15,9 +15,54 @@ def index
 
   def populate
     if current_user.admin?
-      render json: get_table_json_data
+      table = get_table_json_data
+      assignment_marks = []
+      gefms = []
+      markschemes = []
+      table[0][:assignment_marks].each do |key|
```

Comment:
Similar to my comment on the other PR, it is much better to use your own query here. To start try plucking just the `:id` and `:short_identifier` for assignments, and then using a `map` to generate the column data.